### PR TITLE
[Type-definitions] Fix type generator to support import maps

### LINKED ISSUE #16705: [Request for help from TypeScript users] Re-enable `gulp typestest` in GitHub Actions
*PLEASE NOTE:* There's a risk that the type-definitions themselves are also broken now (and not just the tests), which means that unless this is fixed very soon the next PDF.js release may not have working types.

---

This was *temporary* disabled in PR #16698, and should be re-enabled as soon as possible.

0. If you're new to the project, please see https://github.com/mozilla/pdf.js#getting-the-code and https://github.com/mozilla/pdf.js/wiki/Contributing
1. Revert commit https://github.com/mozilla/pdf.js/commit/d9350c38997b31ee9f0058f438b332997eb53e07 locally, using `git revert d9350c3 --no-edit`.
3. Update the following code to handle (at least) the new `display-svg` import:
   https://github.com/mozilla/pdf.js/blob/43fc78899fea71cdadb4883aeeec51b849c29b3d/gulpfile.mjs#L1531-L1548
4. Run `gulp typestest`, and `gulp ci-test`, to validate your changes.
5. Finally, ensure that both `gulp jsdoc` and `gulp web` still work.

### PR DESCRIPTION
**Pull request by stof**
This fixes https://github.com/mozilla/pdf.js/issues/16705 fully by making typescript aware of the import map when generating type declarations.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -7,7 +7,18 @@
...
+    "moduleResolution": "node",
+    "paths": {
+      "display-fetch_stream": ["./src/display/fetch_stream"],
+      "display-l10n_utils": ["./web/l10n_utils"],
+      "display-network": ["./src/display/network"],
+      "display-node_stream": ["./src/display/node_stream"],
+      "display-node_utils": ["./src/display/node_utils"],
+      "display-svg": ["./src/display/svg"],
+      "web-com": ["./web/genericcom"],
+      "web-print_service": ["./web/pdf_print_service"],
```
Comment (Snuffleupagus) [inline tsconfig.json:19]: These two should not be necessary, since they are only used in the default viewer and not actually in the "viewer components".

Comment (stof) [reply tsconfig.json]: I removed them

**Review by Snuffleupagus — APPROVED** [review]
As far as I can tell, not really knowing TypeScript, this seems to work just fine.

r=me, thank you!

**Comment by stof** [comment]
@Snuffleupagus is there any ETA for the next release that would ship non-broken types again ? As I'm new to this project, I don't know how the release schedule is managed.

**Comment by Snuffleupagus** [comment]
Nowadays releases usually happen on the *last Sunday* of every month, unless something unforeseen happens.

**Comment by stof** [comment]
thanks

**Pull request merged**